src/Codeception/Lib/ModuleContainer.php

```diff
@@ -35,207 +61,364 @@ public function __construct(Di $di, $config)
     }
 
     /**
-     * @param $moduleName
+     * Create a module.
+     *
+     * @param string $moduleName
      * @param bool $active
-     * @throws ConfigurationException
-     * @return Module
+     * @return \Codeception\Module
+     * @throws \Codeception\Exception\ConfigurationException
+     * @throws \Codeception\Exception\ModuleException
+     * @throws \Codeception\Exception\ModuleRequireException
+     * @throws \Codeception\Exception\InjectionException
      */
     public function create($moduleName, $active = true)
     {
         $this->active[$moduleName] = $active;
+
+        $moduleClass = $this->getModuleClass($moduleName);
+        if (!class_exists($moduleClass)) {
+            throw new ConfigurationException("Module $moduleName could not be found and loaded");
+        }
+
         $config = $this->getModuleConfig($moduleName);
 
-        // skip config validation on dependent module
         if (empty($config) && !$active) {
+            // For modules that are a dependency of other modules we want to skip the validation of the config.
+            // This config validation is performed in \Codeception\Module::__construct().
+            // Explicitly setting $config to null skips this validation.
             $config = null;
         }
 
-        // helper
-        $hasNamespace = (strpos($moduleName, '\\') !== false);
-        if ($hasNamespace) {
-            return $this->instantiate($moduleName, $moduleName, $config);
+        $this->modules[$moduleName] = $module = $this->di->instantiate($moduleClass, [$this, $config], false);
+
+        if ($this->moduleHasDependencies($module)) {
+            $this->injectModuleDependencies($moduleName, $module);
         }
 
-        // standard module
-        $moduleClass = self::MODULE_NAMESPACE . $moduleName;
-        if (class_exists($moduleClass)) {
-            return $this->instantiate($moduleName, $moduleClass, $config);
+        // If module is not active its actions should not be included in the actor class
+        $actions = $active ? $this->getActionsForModule($module, $config) : [];
+
+        foreach ($actions as $action) {
+            $this->actions[$action] = $moduleName;
+        };
+
+        return $module;
+    }
+
+    /**
+     * Does a module have dependencies?
+     *
+     * @param \Codeception\Module $module
+     * @return bool
+     */
+    private function moduleHasDependencies($module)
+    {
+        if (!$module instanceof DependsOnModule) {
+            return false;
         }
 
-        throw new ConfigurationException("Module $moduleName could not be found and loaded");
+        return !empty($module->_depends());
```

Comment:
Please replace this line with `return count($module->_depends()) > 0;` for compatibility with PHP 5.4